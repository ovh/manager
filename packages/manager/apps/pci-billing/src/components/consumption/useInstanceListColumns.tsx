import {
  DataGridTextCell,
  DateFormat,
  useFormattedDate,
} from '@ovh-ux/manager-react-components';
import { ODS_THEME_COLOR_INTENT } from '@ovhcloud/ods-common-theming';
import {
  ODS_BUTTON_SIZE,
  ODS_BUTTON_TYPE,
  ODS_BUTTON_VARIANT,
} from '@ovhcloud/ods-components';
import {
  OsdsButton,
  OsdsPopover,
  OsdsPopoverContent,
} from '@ovhcloud/ods-components/react';
import { useTranslation } from 'react-i18next';
import { useNavigate } from 'react-router-dom';
import InstanceDetailPopover from './InstanceDetailPopover.component';

export type TInstanceConsumptionDetail = {
  instanceId: string;
  instanceName: string;
  total: string;
  region: string;
  reference: string;
  imageType: string;
  vmType: string;
  isDeleted: boolean;
  monthlyBilling?: {
    since: string;
    status: string;
  };
  planCode?: string;
  monthlyPlan: string[];
};

export type UseInstanceListColumnsProps = {
  colNameLabel: string;
  colTotalLabel: string;
  showAdditionalInstanceDetails?: boolean;
  isMonthlyInstances?: boolean;
};

export function useInstanceListColumns({
  colNameLabel,
  colTotalLabel,
  showAdditionalInstanceDetails = false,
  isMonthlyInstances = false,
}: UseInstanceListColumnsProps) {
  const { t } = useTranslation('consumption/hourly-instance/instance');
  const navigate = useNavigate();
  // TO DISPLAY CONFIRM MONTHLY PAYMENT BUTTON
  const displayUpdateToMonthlyBillingButton = (
    instanceConsumption: TInstanceConsumptionDetail,
  ) =>
    !isMonthlyInstances &&
    instanceConsumption.monthlyBilling === null &&
    !instanceConsumption.isDeleted &&
    instanceConsumption.monthlyPlan;

  return [
    {
      id: 'name',
      cell: (row: TInstanceConsumptionDetail) => (
        <OsdsPopover>
          <div slot="popover-trigger" className="cursor-pointer">
            {row?.instanceName}
          </div>
          <OsdsPopoverContent>
            <InstanceDetailPopover row={row} />
          </OsdsPopoverContent>
        </OsdsPopover>
      ),
      label: colNameLabel,
    },
    {
      id: 'consumption',
      cell: (row: TInstanceConsumptionDetail) => (
        <div>
          <DataGridTextCell>{row.total}</DataGridTextCell>
        </div>
      ),
      label: colTotalLabel,
    },
    (!isMonthlyInstances || showAdditionalInstanceDetails) && {
      id: 'action',
      cell: (row: TInstanceConsumptionDetail) => (
        <div className="min-w-[16rem]">
          {['ok', 'resized'].includes(row.monthlyBilling?.status) && (
            <DataGridTextCell>
              {t('cpbc_hourly_instance_monthly_billing_since', {
                since: useFormattedDate({
                  dateString: row.monthlyBilling.since,
                  format: DateFormat.compact,
                }),
              })}
            </DataGridTextCell>
          )}
          {row.monthlyBilling?.status === 'activationPending' && (
            <DataGridTextCell>
              {t('cpbc_hourly_instance_monthly_billing_activationPending')}
            </DataGridTextCell>
          )}
          {!row.monthlyBilling && row.isDeleted && (
            <DataGridTextCell className="opacity-50">
              {t('cpbc_hourly_instance_deleted')}
            </DataGridTextCell>
          )}

          {displayUpdateToMonthlyBillingButton(row) && (
            <OsdsButton
              color={ODS_THEME_COLOR_INTENT.primary}
              type={ODS_BUTTON_TYPE.button}
              size={ODS_BUTTON_SIZE.sm}
              variant={ODS_BUTTON_VARIANT.stroked}
              onClick={() => navigate(`./monthly?instanceId=${row.instanceId}`)}
            >
              {t('cpbc_hourly_instance_pass_to_monthly')}
            </OsdsButton>
          )}
        </div>
      ),
      label: '',
    },
  ].filter(Boolean);
}
