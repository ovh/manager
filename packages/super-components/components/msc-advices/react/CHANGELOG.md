# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [1.1.1](https://github.com/ovh/manager/compare/@ovhcloud/msc-react-advices@1.1.0...@ovhcloud/msc-react-advices@1.1.1) (2023-12-05)

**Note:** Version bump only for package @ovhcloud/msc-react-advices





# [1.1.0](https://github.com/ovh/manager/compare/@ovhcloud/msc-react-advices@1.0.0...@ovhcloud/msc-react-advices@1.1.0) (2023-12-05)


### Features

* msc-advices ([670db53](https://github.com/ovh/manager/commit/670db53549b433db305fa4e6875d804dc2da56c3))
