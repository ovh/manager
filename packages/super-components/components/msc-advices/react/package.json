{
  "name": "@ovhcloud/msc-react-advices",
  "version": "1.0.0",
  "private": true,
  "description": "React specific wrapper for msc-advices super component",
  "keywords": [],
  "homepage": "https://github.com/ovh/manager/tree/master/packages/super-components/components/msc-advices#readme",
  "bugs": {
    "url": "https://github.com/ovh/manager/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ovh/manager.git",
    "directory": "packages/super-components/components/msc-advices/react"
  },
  "license": "Apache-2.0",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist/"
  ],
  "scripts": {
    "clean": "rimraf dist",
    "compile": "npm run compile:esm && npm run compile:cjs",
    "compile:cjs": "tsc -p tsconfig.cjs.json",
    "compile:esm": "tsc -p tsconfig.json",
    "msc:build": "npm run clean && npm run compile",
    "prepare": "yarn msc:build"
  },
  "dependencies": {
    "@ovhcloud/msc-stencil-advices": "*",
    "tslib": "*"
  },
  "devDependencies": {
    "@types/lodash": "4.14.197",
    "@types/react": "18.2.22",
    "@types/react-dom": "18.2.7",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "rimraf": "^3.0.2",
    "typescript": "5.1.3"
  },
  "peerDependencies": {
    "@ovhcloud/ods-components": "^16.0.1",
    "react": ">=18.0.0",
    "react-dom": ">=18.0.0"
  }
}
