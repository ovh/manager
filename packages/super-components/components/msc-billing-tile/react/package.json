{
  "name": "@ovhcloud/msc-react-billing-tile",
  "version": "1.1.1",
  "private": true,
  "description": "React specific wrapper for Billing Tile super component",
  "keywords": [],
  "homepage": "https://github.com/ovh/manager/tree/master/packages/super-components/components/msc-billing-tile#readme",
  "bugs": {
    "url": "https://github.com/ovh/manager/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ovh/manager.git",
    "directory": "packages/super-components/components/msc-billing-tile/react"
  },
  "license": "BSD-3-Clause",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist/"
  ],
  "scripts": {
    "build": "yarn msc:build",
    "clean": "rimraf dist",
    "compile": "yarn compile:esm && yarn compile:cjs",
    "compile:cjs": "tsc -p tsconfig.cjs.json",
    "compile:esm": "tsc -p tsconfig.json",
    "msc:build": "yarn clean && yarn compile",
    "prepare": "yarn msc:build"
  },
  "dependencies": {
    "@ovhcloud/msc-stencil-billing-tile": "^1.1.1",
    "tslib": "*"
  },
  "devDependencies": {
    "@types/lodash": "4.14.197",
    "@types/react": "18.2.22",
    "@types/react-dom": "18.2.7",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "rimraf": "^3.0.2",
    "typescript": "5.1.3"
  },
  "peerDependencies": {
    "react": ">=18.0.0",
    "react-dom": ">=18.0.0"
  }
}
