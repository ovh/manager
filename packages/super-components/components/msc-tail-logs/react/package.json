{
  "name": "@ovhcloud/msc-react-tail-logs",
  "version": "1.0.0",
  "private": true,
  "description": "React specific wrapper for Tail Logs super components",
  "keywords": [],
  "homepage": "https://github.com/ovh/manager/tree/master/packages/super-components/components/msc-tail-logs#readme",
  "bugs": {
    "url": "https://github.com/ovh/manager/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ovh/manager.git",
    "directory": "packages/super-components/components/msc-tail-logs/react"
  },
  "license": "Apache-2.0",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist/"
  ],
  "scripts": {
    "build": "yarn msc:build",
    "clean": "rimraf dist",
    "compile": "yarn compile:esm && yarn compile:cjs",
    "compile:cjs": "tsc -p tsconfig.cjs.json",
    "compile:esm": "tsc -p tsconfig.json",
    "msc:build": "yarn clean && yarn compile",
    "prepare": "yarn msc:build"
  },
  "dependencies": {
    "@ovhcloud/msc-stencil-tail-logs": "*",
    "tslib": "*"
  },
  "devDependencies": {
    "@types/lodash": "4.14.197",
    "@types/react": "18.2.22",
    "@types/react-dom": "18.2.7",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "rimraf": "^3.0.2",
    "typescript": "5.1.3"
  },
  "peerDependencies": {
    "react": ">=18.0.0",
    "react-dom": ">=18.0.0"
  }
}
