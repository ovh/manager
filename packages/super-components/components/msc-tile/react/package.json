{
  "name": "@ovhcloud/msc-react-tile",
  "version": "1.0.0",
  "private": true,
  "description": "React specific wrapper for Manager super components",
  "keywords": [],
  "license": "Apache-2.0",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist/"
  ],
  "scripts": {
    "build": "npm run clean && npm run compile",
    "clean": "rimraf dist",
    "compile": "npm run compile:esm && npm run compile:cjs",
    "compile:cjs": "tsc -p tsconfig.cjs.json",
    "compile:esm": "tsc -p tsconfig.json"
  },
  "dependencies": {
    "@ovhcloud/msc-stencil-tile": "^1.0.0",
    "tslib": "*"
  },
  "devDependencies": {
    "@types/react": "18.2.8",
    "@types/react-dom": "18.2.4",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "rimraf": "^3.0.2",
    "typescript": "5.1.3"
  },
  "peerDependencies": {
    "@ovhcloud/ods-stencil": "^13.0.1",
    "react": ">=18.0.0",
    "react-dom": ">=18.0.0"
  }
}
