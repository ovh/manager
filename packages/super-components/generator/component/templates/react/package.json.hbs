{
  "name": "@ovhcloud/msc-react-{{ packageName }}",
  "version": "1.0.0",
  "private": true,
  "description": "React specific wrapper for Manager super components",
  "keywords": [],
  "homepage": "https://github.com/ovh/manager/tree/master/packages/super-components/components/{{ componentTag }}#readme",
  "bugs": {
    "url": "https://github.com/ovh/manager/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ovh/manager.git",
    "directory": "packages/super-components/components/{{ componentTag }}/react"
  },
  "license": "Apache-2.0",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist/"
  ],
  "scripts": {
    "prepare": "yarn msc:build",
    "msc:build": "npm run clean && npm run compile",
    "clean": "rimraf dist",
    "compile": "npm run compile:esm && npm run compile:cjs",
    "compile:cjs": "tsc -p tsconfig.cjs.json",
    "compile:esm": "tsc -p tsconfig.json"
  },
  "dependencies": {
    "@ovhcloud/msc-stencil-{{ packageName }}": "*",
    "tslib": "*"
  },
  "devDependencies": {
    "@types/lodash": "4.14.197",
    "@types/react": "17.0.39",
    "@types/react-dom": "18.2.7",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "rimraf": "^3.0.2",
    "typescript": "5.1.3"
  },
  "peerDependencies": {
    "@ovhcloud/ods-stencil": "^14.1.0",
    "react": ">=18.0.0",
    "react-dom": ">=18.0.0"
  }
}
