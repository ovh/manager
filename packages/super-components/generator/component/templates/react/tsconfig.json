{
  "extends": "../../../tsconfig.react.json",
  "compilerOptions": {
    "outDir": "dist/esm",
    "declarationDir": "dist/types"
  },
  "include": ["src/**/*.ts", "src/**/*.tsx"]
}
