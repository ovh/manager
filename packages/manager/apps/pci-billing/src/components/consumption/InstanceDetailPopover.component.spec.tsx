import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import InstanceDetailPopover from './InstanceDetailPopover.component';
import { TInstanceConsumptionDetail } from './useInstanceListColumns';

describe('InstanceDetailPopover', () => {
  const instanceDetail = {
    instanceId: '057ff95a-42df-4678-b574-660e0073c270',
    instanceName: 'InstancedA',
    total: '88.16 â‚¬',
    region: 'GRA-STAGING-A',
    reference: 'a10-45',
    imageType: 'linux',
    vmType: 'A10-45',
    isDeleted: false,
    monthlyBilling: null,
    planCode: null,
  } as TInstanceConsumptionDetail;

  it('matches snapshot', () => {
    const { asFragment } = render(
      <InstanceDetailPopover row={instanceDetail} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
