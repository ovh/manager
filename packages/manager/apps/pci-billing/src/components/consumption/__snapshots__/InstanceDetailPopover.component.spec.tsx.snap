// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InstanceDetailPopover > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col gap-3"
  >
    <osds-text
      class="block"
      color="text"
      level="body"
      size="400"
    >
      <strong>
        cpbc_server_details_popover_region : 
      </strong>
      manager_components_region_A_micro
    </osds-text>
    <osds-text
      class="block"
      color="text"
      level="body"
      size="400"
    >
      <strong>
        cpbc_server_details_popover_image_type : 
      </strong>
      cpbc_server_details_popover_imagetype_linux
    </osds-text>
    <osds-text
      class="block"
      color="text"
      level="body"
      size="400"
    >
      <strong>
        cpbc_server_details_popover_vm_type : 
      </strong>
      A10-45
    </osds-text>
  </div>
</DocumentFragment>
`;
